fuel-merkle/src/sparse/verify.rs

```diff
@@ -0,0 +1,341 @@
+use crate::{
+    common::{
+        path::{
+            Instruction,
+            Path,
+        },
+        Bytes32,
+        Prefix,
+    },
+    sparse::{
+        empty_sum,
+        proof::{
+            ExclusionProof,
+            InclusionProof,
+            Proof,
+        },
+        Node,
+    },
+};
+
+pub fn verify<K: Into<Bytes32>, V: AsRef<[u8]>>(key: K, value: &V, proof: Proof) -> bool {
+    match proof {
+        Proof::InclusionProof(proof) => verify_inclusion(key, value, proof),
+        Proof::ExclusionProof(proof) => {
+            value.as_ref() == empty_sum() && verify_exclusion(key, proof)
```

Comment:
The `value.as_ref() == empty_sum()` check truly is not required. It is more of a hack to have one function for both poofs.

Reply:
If the value is not the empty sum, then we already know that the exclusion proof should fail. 

I'm not sure I agree that this is a hack, but I can that check into `verify_exclusion` if you prefer. I can also make the `verify_*` functions into member functions on the `*Proof` structs themselves.